src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].every((side) => side <= 0)) {
+      throw new Error('Sides must be bigger then 0');
+    }
+
+    if (a + b <= c || a + b <= b || b + c <= a) {
+      throw new Error('The sides provided do not form a valid triangle');
+    }
+  }
```

Comment:
you can sort you array and add only 1 check
```ts
const sides = [];

    sides.push(this.a, this.b, this.c);

    sides.sort((side1, side2) => side2 - side1);

    if (sides[0] >= sides[1] + sides[2]) {
      throw new Error(
        'error message',
      );
    }
```
but yout code shorter, so it`s ok